schema/omega-services.async.api.yaml

```diff
@@ -1,525 +1,480 @@
+        (i.e. that it is valid JSON/XML/RDF/Binary etc.)
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
       payload:
-        $ref: "#/components/schemas/editSet"
-      correlationId:
-        description: Default Correlation ID
-        location: $message.header#/correlationId
-
-    editSetRecord:
-      name: editSetRecord
-      title: EditSet Record
-      summary: An Edit Set Record in the Omega Catalogue
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+
+    InvalidApplicationError:
+      messageId: OELIFERR003
+      name: InvalidApplicationError
+      title: Invalid Application Error
+      description: >-
+        The client application as identified by the `OMGApplicationID` header is
+        either invalid, not-recognised, or prohibited.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
       payload:
-        $ref: "#/components/schemas/editSetRecord"
-      correlationId:
-        description: Default Correlation ID
-        location: $message.header#/correlationId
-
-    updateResponseStatus:
-      name: updateResultStatus
-      title: Update Result Status
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+
+    AuthenticationError:
+      messageId: OELIFERR004
+      name: AuthenticationError
+      title: Authentication Error
+      description: >-
+        Authentication failed, likely due to an invalid token being present in
+        the `OMGToken` header.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
       payload:
-        $ref: "#/components/schemas/updateResponseStatus"
-      correlationId:
-        description: Default Correlation ID
-        location: $message.header#/correlationId
-
-    errorResponse:
-      name: errorResponse
-      title: Error response message
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
+
+    UnrecognisedMessageTypeError:
+      messageId: OELIFERR005
+      name: UnrecognisedMessageTypeError
+      title: Unrecognised Message Type Error
+      description: >-
+        The value of `OMGMessageTypeID` header was not recognised and so the
+        message cannot be further dispatched for processing.
+      traits:
+        - $ref: '#/components/messageTraits/JMSMessage'
+        - $ref: '#/components/messageTraits/JMSResponse'
+        - $ref: '#/components/messageTraits/OMGMessage'
+      payload:
+        type: array
+        items:
+          $ref: '#/components/schemas/ErrorMessage'
```

Comment:
Do we really need this as a separate error type? Validating the `OMGMessageTypeID` header can be done as part of the initial validation and if it is not recognised we return an `OELIFERR002`. We can provide more detail in the error body but it just makes more work trying to handle this separately.

Reply:
These values should come from a controlled vocabulary that we publish. If this header has an invalid value then all bets are off. If it is missing then that would instead be a `OELIFERR002`

Reply:
Could we not just say if it is invalid _or missing_ return `OELIFERR003` as it will naturally fail validation if it is missing.

Reply:
Checking for required headers happens at the `OELIFERR002` stage, checking their values happens after that.

Reply:
Sure we can do it after. The initial implementation checked all these things at once. It just seems slightly inefficient to delay checking that the service exists until the next stage - but if we want to use a different message ID for this then that is what we have to do.